ee/clickhouse/views/test/test_experiment_saved_metrics.py

```diff
@@ -0,0 +1,131 @@
+from rest_framework import status
+
+from ee.api.test.base import APILicensedTest
+from posthog.models.experiment import Experiment, ExperimentToSavedMetric
+
+
+class TestExperimentSavedMetricsCRUD(APILicensedTest):
+    def test_can_list_experiment_saved_metrics(self):
+        response = self.client.get(f"/api/projects/{self.team.id}/experiment_saved_metrics/")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_create_update_experiment_saved_metrics(self) -> None:
+        response = self.client.post(
+            f"/api/projects/{self.team.id}/experiment_saved_metrics/",
+            data={
+                "name": "Test Experiment saved metric",
+                "description": "Test description",
+                "query": {"events": [{"id": "$pageview"}]},
+            },
+            format="json",
+        )
+
+        saved_metric_id = response.json()["id"]
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.json()["name"], "Test Experiment saved metric")
+        self.assertEqual(response.json()["description"], "Test description")
+        self.assertEqual(
+            response.json()["query"],
+            {"events": [{"id": "$pageview"}]},
+        )
+        self.assertEqual(response.json()["created_by"]["id"], self.user.pk)
+
+        # Generate experiment to have saved metric
+        ff_key = "a-b-tests"
+        response = self.client.post(
+            f"/api/projects/{self.team.id}/experiments/",
+            {
+                "name": "Test Experiment",
+                "description": "",
+                "start_date": "2021-12-01T10:23",
+                "end_date": None,
+                "feature_flag_key": ff_key,
+                "parameters": None,
+                "filters": {
+                    "events": [
+                        {"order": 0, "id": "$pageview"},
+                        {"order": 1, "id": "$pageleave"},
+                    ],
+                    "properties": [],
+                },
+                "saved_metrics_ids": [{"id": saved_metric_id, "metadata": {"type": "secondary"}}],
```

Comment:
@jurajmajerik currently this is the place i want feedback on - anything else you think we'd need for saved metrics?

I'm unclear on the shape of `metrics` on experiments - what all extra data do they have other than the query? -- and what all metadata will saved metrics need so `all_metrics` look no different whether its custom or saved metrics.

Reply:
I can't think of anything else we would need besides `type`

Currently the shape of `metrics` is:
```
{
    "type": "primary" | "secondary",
    "query": ExperimentTrendsQuery | ExperimentFunnelsQuery
}
```

Reply:
for future expansion, we can either keep a metadata json, or add columns for every new metadata field we need (like metric sets for example). Any preference? I don't see us ever querying for the metadata separately so i don't mind it being json

Reply:
indeed, makes sense 👍